### PR TITLE
Add AbortSignal.any

### LINKED ISSUE #30095: The AbortSignal.any static method not documented
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal

### What specific section or headline is this issue about?

Static methods

### What information was incorrect, unhelpful, or incomplete?

`AbortSignal` has a `any` static method allowing to combine several other signals. This method is not documented (except in the browser compatibility table because BCD has data for it, but it links to a non-existent page).

### What did you expect to see?

I would expect this method to be documented.

### Do you have any supporting links, references, or citations?

https://chromestatus.com/feature/5202879349522432

### Do you have anything more you want to share?

_No response_

### PR DESCRIPTION
**Pull request by Jamesernator**
### Description

This adds MDN docs for the [AbortSignal.any](https://dom.spec.whatwg.org/#dom-abortsignal-any).

### Motivation

This is a new API.

### Additional details

Spec is [here](https://dom.spec.whatwg.org/#dom-abortsignal-any). A change has been made to an example on the main `AbortSignal` page also.

# Note

I'm unsure of where to add the relevant `{{Specifications}}` link. Checking the page the local server just shows `{{Specifications}}` where the specifications box should be (other pages seem to work fine).

Fixes #30095

### CODE REVIEW

**Comment by hamishwillee** [comment]
@Jamesernator MDN document things once there is a browser with an release implementation - existence in a specification is not enough. Do you know of a browser that supports this?

The reason the specifications and browser compatibility are not working is that there is no front matter key like:
```
browser-compat: api.AbortSignal.any_static
```

The browser compatibility data needs to be updated with the same key and data for the supported browser versions: https://github.com/mdn/browser-compat-data/blob/main/api/AbortSignal.json

**Comment by Jamesernator** [comment]
Ah yes this is new in Chrome Dev, admittedly I assumed it was already shipping in browsers as Node has already shipped it.

**Comment by hamishwillee** [comment]
@Jamesernator Actually it being in node/deno is good enough to justify doing the docs, but we should also do the browser compatibility data too.

**Comment by bsmth** [comment]
Hi @Jamesernator - just dropping a note that there are some merge conflicts that need to be resolved here if you'd like to come back to this.

__notes:__
- https://github.com/shaseley/abort-signal-any (explainer)

**Comment by Jamesernator** [comment]
The merge conflict is resolved, I've also added the compat data in [a PR to browser-compat-data](https://github.com/mdn/browser-compat-data/pull/20692).

**Comment by skyclouds2001** [comment]
the issue https://github.com/mdn/content/issues/30095 suggest this contribute, maybe it is a good idea to add a `fixes` to the issue?

File: files/en-us/web/api/abortsignal/any_static/index.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  timeoutSignal,
+]);
+
+try {
+  const res = await fetch(someUrlToDownload, {
+    // Stop the fetch when any of the
+    signal: combinedSignal,
+  });
+  const body = await res.blob();
+  // do something with downloaded content
```
Comment (teoli2003) [inline files/en-us/web/api/abortsignal/any_static/index.md:58]:
```suggestion
  // Do something with downloaded content:
  // …
```

File: files/en-us/web/api/abortsignal/index.md [hunk 2]
```diff
@@ -24,6 +24,8 @@ _The AbortSignal interface may also inherit properties from its parent interface
 
 - {{domxref("AbortSignal/abort_static", "AbortSignal.abort()")}}
   - : Returns an **`AbortSignal`** instance that is already set as aborted.
+- {{domxref("AbortSignal.any()")}}
```
Comment (teoli2003) [inline files/en-us/web/api/abortsignal/index.md:27]:
```suggestion
- {{domxref("AbortSignal/any_static", "AbortSignal.any()")}}
```

File: files/en-us/web/api/abortsignal/any_static/index.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+### Return value
+
+An `AbortSignal` instance with the {{domxref("AbortSignal.aborted")}} property set to `true`, and {{domxref("AbortSignal.reason")}} set to the specified or default reason value.
+
+## Examples
+
+### Using AbortSignal.any()
+
+This example demonstrates combining both a signal from an {{domxref("AbortController")}}, and a timeout signal from {{domxref("AbortSignal.timeout")}}.
```
Comment (teoli2003) [inline files/en-us/web/api/abortsignal/any_static/index.md:32]:
```suggestion
This example demonstrates combining both a signal from an {{domxref("AbortController")}}, and a timeout signal from {{domxref("AbortSignal/timeout_statis", "AbortSignal.timeout")}}.
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
A few details and this is good to go. Good job!

**Comment by teoli2003** [comment]
I have added the "Fixes #xyz" in the description.

File: files/en-us/web/api/abortsignal/any_static/index.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+### Parameters
+
+- `iterable`
+  - : An [iterable](/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#the_iterable_protocol) (such as an {{jsxref("Array")}}) of abort signals.
+
+### Return value
+
+An `AbortSignal` instance with the {{domxref("AbortSignal.aborted")}} property set to `true`, and {{domxref("AbortSignal.reason")}} set to the specified or default reason value.
```
Comment (skyclouds2001) [inline files/en-us/web/api/abortsignal/any_static/index.md:26]:
```suggestion
An `AbortSignal` instance with the {{domxref("AbortSignal.reason", "reason")}} property set to the specified or default reason value.
```

note that `AbortSignal.aborted` should initial to be `false`, only when any of the passing `signals` is already aborted, the `AbortSignal.aborted` will be `true`

File: files/en-us/web/api/abortsignal/any_static/index.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+## Examples
+
+### Using AbortSignal.any()
```
Comment (skyclouds2001) [inline files/en-us/web/api/abortsignal/any_static/index.md:30]:
```suggestion
### Using `AbortSignal.any()`
```

File: files/en-us/web/api/abortsignal/any_static/index.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+This example demonstrates combining both a signal from an {{domxref("AbortController")}}, and a timeout signal from {{domxref("AbortSignal.timeout")}}.
+
+```js
+const button = document.getElementById("cancelDownloadButton");
+
+const userCancelController = new AbortController();
+cancelDownloadButton.addEventListener("click", () => {
+  userCancelController.abort();
+});
```
Comment (skyclouds2001) [inline files/en-us/web/api/abortsignal/any_static/index.md:40]:
```suggestion
const cancelDownloadButton = document.getElementById("cancelDownloadButton");

const userCancelController = new AbortController();
cancelDownloadButton.addEventListener("click", () => {
  userCancelController.abort();
});
```

the `button` should be `cancelDownloadButton`

File: files/en-us/web/api/abortsignal/any_static/index.md (hunk 4 above)
Comment (Jamesernator) [reply files/en-us/web/api/abortsignal/any_static/index.md]: Yes this section is was nonsense, I copied `AbortSignal.abort` as a template and forgot to change this section.

I've updated this section to be similar to [Promise.any](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/any) instead.

**Comment by skyclouds2001** [comment]
Just a usage suggestion to Github, when others give a review, you can click the `commit suggestion` button to directly commit the review to your pull request, rather then do the commit yourself instead

image

also, you can go to the `files` panel and use the `add suggestion to batch` button to merge multpile reviews in a single commit

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**